File: app/src/main/java/com/kipucode/data/local/dao/LessonDao.java
package com.kipucode.data.local.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.kipucode.data.local.dao.relation.LessonWithExercises;
import com.kipucode.data.local.model.Lesson;

import java.util.List;

@Dao
public interface LessonDao {
    @Insert
    void insert(Lesson lesson);
    @Insert
    void insertAll(List<Lesson> lessons);
    @Query("SELECT * FROM lessons")
    List<Lesson> getAllLessons();
    @Query("SELECT * FROM lessons")
    List<LessonWithExercises> getLessonWithExercises();
}

File: app/src/main/java/com/dam/kipucode/data/dao/LessonDao.kt
package com.dam.kipucode.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.jg.kipucode.data.local.dao.relation.LessonWithExercises
import com.jg.kipucode.data.local.model.Lesson

@Dao
interface LessonDao {
    @Insert
    fun insert(lesson: Lesson)

    @Insert
    fun insertAll(lessons: List<Lesson>)

    @Query("SELECT * FROM lessons")
    fun getAllLessons(): List<Lesson>

    @Query("SELECT * FROM lessons")
    fun getLessonWithExercises(): List<LessonWithExercises>
}
